docs/ref/filters.txt

```diff
@@ -3,6 +3,118 @@ Filter Reference
 
 This is a reference document with a list of the filters and their arguments.
 
+.. _core-arguments:
+
+Core Arguments
+--------------
+
+The following are the core arguments that apply to all filters.
+
+``name``
+~~~~~~~~
+
+The name of the field this filter is supposed to filter on, if this is not
+provided it automatically becomes the filter's name on the ``FilterSet``.
+You can traverse "relationship paths" using Django's ``__`` syntax to filter
+fields on a related model. eg, ``manufacturer__name``.
+
+``label``
+~~~~~~~~~
+
+The label as it will apear in the HTML, analogous to a form field's label
+argument.
+
+``widget``
+~~~~~~~~~~
+
+The django.form Widget class which will represent the ``Filter``.  In addition
+to the widgets that are included with Django that you can use there are
+additional ones that django-filter provides which may be useful:
+
+    * ``django_filters.widgets.LinkWidget`` -- this displays the options in a
+      manner similar to the way the Django Admin does, as a series of links.
+      The link for the selected option will have ``class="selected"``.
+
+``action``
```

Comment:
Although, it could actually be useful. A `NumberFilter` would perform the value validation while still passing off the actual filtering to `action`. Currently, `MethodFilter`s do not perform any field validation (would need to either subclass with a custom `field_class` or perform validation in the action itself.

I'd have to look into it further, but it might make sense to deprecate `MethodFilter` and merge the functionality back into the base filter class as just specifying an `action`. Take a look at the drf-filters implementation [here](https://github.com/philipn/django-rest-framework-filters/blob/master/rest_framework_filters/filters.py#L84-L125). It might need a little work, but the `resolve_action` hook could be called after the name is set [here](https://github.com/carltongibson/django-filter/blob/develop/django_filters/filterset.py#L43-L44)